### ISSUE TITLE
[Bug]: Owner is PUBLIC role in databases and tables in non-sys account

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#8412]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:

In a new created account:

select relname,owner from mo_tables;
+----------------------------+-------+
| relname                    | owner |
+----------------------------+-------+
| %!%mo_increment_columns    |     1 |
| mo_user                    |     1 |
| mo_role                    |     1 |
| mo_user_grant              |     1 |
| mo_role_grant              |     1 |
| mo_role_privs              |     1 |
| mo_user_defined_function   |     1 |
| mo_mysql_compatbility_mode |     1 |
| %!%mo_increment_columns    |     1 |
| statement_info             |     1 |
| %!%mo_increment_columns    |     1 |
| metric                     |     1 |
| sql_statement_total        |     1 |
| sql_statement_errors       |     1 |
| sql_transaction_total      |     1 |
| sql_transaction_errors     |     1 |
| server_connections         |     1 |
| server_storage_usage       |     1 |
| %!%mo_increment_columns    |     1 |
| key_column_usage           |     1 |
| columns                    |     1 |
| profiling                  |     1 |
| PROCESSLIST                |     1 |
| user_privileges            |     1 |
| schemata                   |     1 |
| character_sets             |     1 |
| triggers                   |     1 |
| tables                     |     1 |
| views                      |     1 |
| engines                    |     1 |
| routines                   |     1 |
| parameters                 |     1 |
| keywords                   |     1 |
| %!%mo_increment_columns    |     1 |
| user                       |     1 |
| db                         |     1 |
| procs_priv                 |     1 |
| columns_priv               |     1 |
| tables_priv                |     1 |
| mo_columns                 |     0 |
| mo_database                |     0 |
| mo_tables                  |     0 |


 select * from mo_role;
+---------+--------------+---------+-------+---------------------+----------+
| role_id | role_name    | creator | owner | created_time        | comments |
+---------+--------------+---------+-------+---------------------+----------+
|       2 | accountadmin |       1 |     0 | 2023-03-13 09:38:03 |          |
|       1 | public       |       1 |     0 | 2023-03-13 09:38:03 |          |
+---------+--------------+---------+-------+---------------------+----------+
```

### Actual Behavior

the owner should be accountadmin, which role_id is 2

**Comment by DanielZhangQD** [comment]
To be fixed by end of 3.16

**Comment by daviszhen** [comment]
it was fixed, I think.
